Update Gen3 Renesas programming procedure

abinayaddhandapani opened the pull request:
Update the Renesas programming procedure as per the RENESAS GENERATION 3.5 PROGRAMMING GUIDE.

Update the Infineon TDA programming procedure as per the AN_2308_163442-V1.2(TDA38xxxRev5 Programming Guide)

Add PMBUS address and Slave address in the Congo and Morocco config file.

Test fields: Verified in Congo system

Commit:
Update Gen3 Renesas programming procedure

Update the programming procedure as per the
RENESAS GENERATION 3.5 PROGRAMMING GUIDE.

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

ojayanth commented on inc/vr_update_renesas_gen3.hpp:
```diff
@@ -10,12 +10,12 @@
 #define DMA_WRITE             0xC7
 #define DMA_READ             (0xC5)
-#define GEN3_CRC_ADDR        0x0094
+#define GEN3_CRC_ADDR        (0x00F8)
```
Why these Addresses are getting changed? is it specific to version change ?

abinayaddhandapani replied:
These are the new registers to be accessed for the new parts in Venice

Commit:
Update Infineon TDA programming procedure

Update the programming procedure as per the
AN_2308_163442-V1.2(TDA38xxxRev5 Programming Guide)

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

Commit:
Add PMBUS and Slave address for TDA VR's

Add PMBUS address and Slave address in the Congo
and Morocco config file.

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

supven01 commented on src/vr_update_infineon_tda.cpp:
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
                             if(index_range < INDEX_40)
                                 break;
-                            else if(index_range > INDEX_70 && index_range < INDEX_200)
+                            else if(index_range >= 0x80 && index_range < INDEX_200)
```
magic index range 0x80

abinayaddhandapani replied:
Ack

supven01 commented on src/vr_update_infineon_tda.cpp:
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
+                            else if(index_range >= 0x80 && index_range < INDEX_200)
                                 break;
-                            else if(index_range > INDEX_2FF)
+                            else if(index_range >= 0x300 && index_range < 0x380)
```
magic numbers

abinayaddhandapani replied:
Ack

supven01 commented on src/vr_update_infineon_tda.cpp:
```diff
@@ -279,9 +250,11 @@ bool vr_update_infineon_tda::UpdateFirmware()
-                            else if(index_range > INDEX_2FF)
+                            else if(index_range >= 0x300 && index_range < 0x380)
+                                break;
+                            else if(index_range >= 0x390)
```
magic number

abinayaddhandapani replied:
Ack
